File: check_brackets/src/check_brackets.py
import sys, functools
from collections import namedtuple

Bracket = namedtuple("Bracket", ["char", "position"])


def are_matching(left, right):
    return (left + right) in ["()", "[]", "{}"]


def find_mismatch(text):
    text = str(text)
    opening_brackets_stack = []
    for i, next in enumerate(text):
        if next in "([{":
            # Process opening bracket, write your code here
            opening_brackets_stack.append(next)
        if next in ")]}":
            # Process closing bracket, write your code here
            pop = opening_brackets_stack[-1]
            if next == ")" and pop == "(":
                opening_brackets_stack.pop()
            elif next == "}" and pop == "{":
                opening_brackets_stack.pop()
            elif next == "]" and pop == "[":
                opening_brackets_stack.pop()
            else: 
                return i
    return "Success"

def main():
    text = sys.stdin.read()
    ret = find_mismatch(text)
    # Printing answer, write your code here
    print ret

MAINFUNC = main
###########################################################################################
import traceback

if __name__ == '__main__':
  try:
    ret = MAINFUNC()
  except Exception as e:
    details = traceback.format_exc()
    print "\nRuntime error: %s\n\n %s"%(e, details)
  else:
    pass

File: check_brackets/src/check_brackets.cpp
#include <iostream>
#include <stack>
#include <string>

struct Bracket {
    Bracket(char type, int position):
        type(type),
        position(position)
    {}

    bool Matchc(char c) {
        if (type == '[' && c == ']')
            return true;
        if (type == '{' && c == '}')
            return true;
        if (type == '(' && c == ')')
            return true;
        return false;
    }

    char type;
    int position;
};

int main() {
    std::string text;
    getline(std::cin, text);

    bool ret = true;
    int ret_num = 0;
    std::stack <Bracket> opening_brackets_stack;
    for (int position = 0; position <(int) text.length(); ++position) {
        char next = text[position];

        if (next == '(' || next == '[' || next == '{') {
            // Process opening bracket, write your code here
            opening_brackets_stack.push(Bracket(next, position));
        }

        if (next == ')' || next == ']' || next == '}') {
            // Process closing bracket, write your code here
            Bracket b = opening_brackets_stack.top();
            opening_brackets_stack.pop();
            if(!b.Matchc(next)){
                ret_num = position;
                ret = false;
                break;
            }
        }
    }
    if(ret)
        std::cout<<"Success"<<std::endl;
        
    else
        std::cout<<ret_num<<std::endl;
    // Printing answer, write your code here

    return 0;
}
